installer/aws_cluster.go

```diff
@@ -0,0 +1,605 @@
+package installer
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/awslabs/aws-sdk-go/aws"
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/awslabs/aws-sdk-go/gen/cloudformation"
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/awslabs/aws-sdk-go/gen/ec2"
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/awslabs/aws-sdk-go/gen/route53"
+	"github.com/flynn/flynn/pkg/awsutil"
+	"github.com/flynn/flynn/pkg/sshkeygen"
+	"github.com/flynn/flynn/util/release/types"
+)
+
+var DisallowedEC2InstanceTypes = []string{"t1.micro", "t2.micro", "t2.small", "m1.small"}
+var DefaultInstanceType = "m3.medium"
+var StackNotFoundError = errors.New("Stack does not exist")
+
+func (c *AWSCluster) Base() *BaseCluster {
+	return c.base
+}
+
+func (c *AWSCluster) wrapRequest(runRequest func() error) error {
+	const rateExceededErrStr = "Rate exceeded"
+	const maxBackoff = 10 * time.Second
+	backoff := 1 * time.Second
+	for {
+		err := runRequest()
+		if err != nil && err.Error() == rateExceededErrStr {
+			time.Sleep(backoff)
+			backoff = backoff * 2
+			if backoff > maxBackoff {
+				backoff = maxBackoff
+			}
```

Comment:
Testing this with around 6 parallel installs, the backoff rarely exceeded `8s` (mostly `2s` and `4s`). Does a timeout after 34 seconds (5 retries) sound reasonable?